### PR TITLE
feat: Change solving interface

### PR DESCRIPTION
**Pull request by thofma**
- Use the `Solve.*` functionality
- Hide all previous `*solve*` and `*kernel*` functions using
  underscores

[breaking]

### CODE REVIEW

File: src/Matrix.jl [hunk 1]
```diff
@@ -3646,164 +3646,164 @@ function can_solve_left_reduced_triu(r::MatElem{T},
...
+     _can_solve(a::MatElem{S}, b::MatElem{S}; side::Symbol = :right) where S <: RingElement
+ 
+ Given two matrices $a$ and $b$ over the same ring, check the solubility
+ of $ax = b$ if `side` is `:right` or $xa = b$ if `side` is `:left`.
+ Return true if a solution exists, false otherwise. If the dimensions
+ of $a$ and $b$ are incompatible, an exception is raised. If a solution
+ should be computed as well, use `can_solve_with_solution` instead.
+ """
+ function _can_solve(a::MatElem{S}, b::MatElem{S}; side::Symbol = :right) where S <: RingElement
+    return can_solve_with_solution(a, b; side=side)[1]
```
Comment (lgoettgens) [inline src/Matrix.jl:3805]: In this file, there are a lot of weird whitespace additions. Are those intended?

File: test/runtests.jl [hunk 2]
```diff
@@ -3,7 +3,8 @@ using AbstractAlgebra
 using Test
 
 
-if VERSION >= v"1.8.0"
+# disable until we encounter GC problems again
+if false VERSION >= v"1.8.0"
```
Comment (lgoettgens) [inline test/runtests.jl:7]:
```suggestion
if false # VERSION >= v"1.8.0"
```

File: src/Solve.jl [hunk 3]
```diff
@@ -286,12 +286,15 @@ end
...
 
-Return a matrix $K$ whose rows give a basis for the left kernel of $A$, that
+Return a matrix $K$ whose rows generate the left kernel of $A$, that
 is, $KA$ is the zero matrix.
 
-If `side == :right`, the columns of $K$ give a basis for the right kernel of $A$, that
+If `side == :right`, the columns of $K$ generate the right kernel of $A$, that
 is, $AK$ is the zero matrix.
 
+If the base ring is a principal ideal domain, the rows or rows respectively of $K$
```
Comment (joschmitt) [inline src/Solve.jl:295]:
```suggestion
If the base ring is a principal ideal domain, the rows or columns respectively of $K$
```

File: test/runtests.jl (hunk 2 above)
Comment (thofma) [reply test/runtests.jl:7]: Does not matter

**Pull request merged**